### PR TITLE
[MTDSA-6163] list foreign properties models

### PR DESCRIPTION
**Pull request by james-work-account**

### CODE REVIEW

File: app/v1/models/request/listForeignPropertes/ListForeignPropertiesRequest.scala [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.models.request.listForeignPropertes
+
+import uk.gov.hmrc.domain.Nino
+
+case class ListForeignPropertiesRequest(nino: Nino, businessId: String, fromDate: String, toDate: String)
```
Comment (Mbrookesy) [inline app/v1/models/request/listForeignPropertes/ListForeignPropertiesRequest.scala:21]: I believe the dates should be optional

File: app/v1/models/response/listForeignProperties/ListForeignPropertiesResponse.scala [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  implicit object LinksFactory extends HateoasListLinksFactory[ListForeignPropertiesResponse, SubmissionPeriod, ListForeignPropertiesHateoasData] {
+    override def links(appConfig: AppConfig, data: ListForeignPropertiesHateoasData): Seq[Link] = {
+      Seq(
+        listForeignProperties(appConfig, data.nino, data.businessId)
+      )
+    }
+
+    override def itemLinks(appConfig: AppConfig, data: ListForeignPropertiesHateoasData, item: SubmissionPeriod): Seq[Link] =
+      Seq(retrieveForeignProperty(appConfig, data.nino, data.businessId, item.submissionId, rel = RETRIEVE_PROPERTY_PERIOD_SUMMARY))
```
Comment (Mbrookesy) [inline app/v1/models/response/listForeignProperties/ListForeignPropertiesResponse.scala:42]: The Hateoas glossary says that there should also be a create link, it seems to be missing here

File: test/v1/models/response/listForeignProperties/ListForeignPropertiesResponseSpec.scala [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+    "expose the correct item level links for list" in {
+      MockedAppConfig.apiGatewayContext.returns("my/context").anyNumberOfTimes
+      ListForeignPropertiesResponse.LinksFactory.itemLinks(mockAppConfig, ListForeignPropertiesHateoasData(nino, businessId),
+        SubmissionPeriod(submissionId, "", "")) shouldBe
+        Seq(
+          Link(s"/my/context/$nino/$businessId/period/$submissionId", GET, "retrieve-property-period-summary")
+        )
+    }
+  }
```
Comment (Mbrookesy) [inline test/v1/models/response/listForeignProperties/ListForeignPropertiesResponseSpec.scala:146]: Change it to reflect having create included

File: app/v1/models/request/listForeignPropertes/ListForeignPropertiesRequest.scala (hunk 1 above)
Comment (james-work-account) [reply app/v1/models/request/listForeignPropertes/ListForeignPropertiesRequest.scala:21]: Dates are optional in the raw data but we default the dates if they're not provided, so dates will always be sent down to DES.

Comment (Mbrookesy) [reply app/v1/models/request/listForeignPropertes/ListForeignPropertiesRequest.scala:21]: Gotcha

**Pull request merged**